### PR TITLE
feat: add item rarity to loot notifications for monsters

### PR DESCRIPTION
**Pull request by iProdigy**
Partially addresses #139; supersedes #141

image

Resolves:
* https://github.com/BossHuso/discord-rare-drop-notificater/issues/29
* https://github.com/BossHuso/discord-rare-drop-notificater/issues/58
* https://github.com/BossHuso/discord-rare-drop-notificater/issues/82
* https://github.com/BossHuso/discord-rare-drop-notificater/issues/85

Not resolved:
* https://github.com/BossHuso/discord-rare-drop-notificater/issues/26

### CODE REVIEW

File: src/test/java/dinkplugin/util/RarityServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    }
+
+    @Test
+    @DisplayName("Ensure accurate rarity for common drops")
+    void testCommon() {
+        test("Bree", ItemID.SNAPE_GRASS, 5, 7.0 / 127);
+    }
+
+    @Test
+    @Disabled("[BUG] doesn't distinguish by combat level: https://oldschool.runescape.wiki/w/Goblin#Drop_table_2")
```
Comment (iProdigy) [inline src/test/java/dinkplugin/util/RarityServiceTest.java:109]: similar known bug: some scurrius drop rates are doubled since we don't distinguish between private vs. public kills https://oldschool.runescape.wiki/w/Scurrius

Comment (iProdigy) [reply src/test/java/dinkplugin/util/RarityServiceTest.java:109]: similarly: green dragon gem drop table rates are doubled (level 79 vs. wildy slayer cave) https://oldschool.runescape.wiki/w/Green_dragon

Comment (iProdigy) [reply src/test/java/dinkplugin/util/RarityServiceTest.java:109]: A possible mitigation: ignore duplicate drop entries (i.e., item id, quantity range, and rarity is same)

should address more edge cases than it introduces if yall agree @pajlada @Felanbird

Comment (iProdigy) [reply src/test/java/dinkplugin/util/RarityServiceTest.java]: Done in https://github.com/pajlads/DinkPlugin/pull/425/commits/aa389a505de560d8bc1246408a594ac6a70eeca7

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ## Unreleased
 
+- Minor: Add item rarity to NPC loot notifications. (#425)
```
Comment (pajlada) [inline CHANGELOG.md:3]: Major imo

**Review by pajlada — COMMENTED** [review]
Would be nice to include the npc_drops.json source, both as a way for us to update it going forward & as a way to attribute the data

**Comment by iProdigy** [comment]
Clarified data source in https://github.com/pajlads/DinkPlugin/pull/425/commits/fb477f0e94c17e277fcaa121c9403bd68bcff9ae

In particular, we feed [their parsed data](https://github.com/Flipping-Utilities/osrs-datasets/blob/js-set/data/monsters/all-monsters.json) to [our script](https://gist.github.com/iProdigy/703afc2ce881728051d20042258bf936) to generate `npc_drops.json` - will eventually clean this up so it can be ran as an action to automatically update our json file

File: README.md [hunk 3]
```diff
@@ -276,3 +276,6 @@ On login, Dink can submit a character summary containing data that spans multipl
 ## Credits
 
 This plugin uses code from [Universal Discord Notifier](https://github.com/MidgetJake/UniversalDiscordNotifier).
+
+Item rarity data is sourced from the OSRS Wiki (licensed under [CC BY-NC-SA 3.0](https://creativecommons.org/licenses/by-nc-sa/3.0/)),
+which was conveniently parsed by [Flipping Utilities](https://github.com/Flipping-Utilities/parsed-osrs) (and transformed by pajlads).
```
Comment (pajlada) [inline README.md:281]: Could we include the transformation util as a utility in this repo, or make a new repo for it & link to it?
I can create it under pajlads if you don't have access

Comment (iProdigy) [reply README.md]: Done in https://github.com/pajlads/DinkPlugin/pull/425/commits/b07ea224c585ce15482b4685cfa9ac1c4a9ff1a9

We can now run `./gradlew generateResources` to update `npc_drops.json` - a separate PR should be created to regularly execute this task from a gh action

**Review by pajlada — APPROVED** [review]
